pkg/logql/engine.go

```diff
@@ -203,10 +203,11 @@ func (q *query) Exec(ctx context.Context) (logqlmodel.Result, error) {
 	log, ctx := spanlogger.New(ctx, "query.Exec")
 	defer log.Finish()
 
+	queryHash := hashedQuery(q.params.Query())
```

Comment:
I think we should verify that the same hash is produced in metrics.go vs this executing query line for queries like:

```
sum(rate(...)) by (label)
```

This query will show up in metrics.go as:

```
sum by (label) (rate(...))
```

And i'm not exactly sure where this re-write happens?

Reply:
Good catch. I pushed a test that checks whether the hashed query emitted by engine vs metrics.go are equal or not. Apparently, `sum(...) by(label)` resulted in the same query hash in both places so not sure I'm missing something :thinking: 

if you know of other interesting queries for me to test I can add them as new test cases in the scenario I implemented on `engine_test.go`

Reply:
ok, no worries! thanks for checking, my recollection here is likely missing something.